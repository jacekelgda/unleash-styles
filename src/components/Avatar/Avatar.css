.AvatarButton {
	display: none;
}
.AvatarButtonLabel {
	display: inline-block;
	position: relative;
	font: 700 11px/36px 'Montserrat', sans-serif;
	text-align: center;
	width: 36px;
	height: 36px;
	border-radius: 50%;
	background-color: #eeddfc;
	color: #8e15fa;
	cursor: pointer;
}
.AvatarButtonLabel::after {
	content: '';
	position: absolute;
	right: -18px;
	top: calc(50% - 2px);
	width: 0;
	height: 0;
	border-style: solid;
	border-width: 5px 4px 0 4px;
	border-color: #eeddfc transparent transparent transparent;
}

.AvatarButton:checked {
	display: block;
}
