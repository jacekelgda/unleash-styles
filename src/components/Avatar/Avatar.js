import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from './Avatar.css'

/**
 * Avatar component which can be used in relation to user profile
 */
class Avatar extends Component {
  static defaultProps: Object

  handleClick () {
    const { onClick } = this.props
    onClick()
  }

  render () {
    const { label } = this.props
    return (
      <div>
        <input
          type='checkbox'
          onClick={this.handleClick}
          className={styles.AvatarButton}
          id='avatar_button'
        />
        <label
          className={styles.AvatarButtonLabel}
          for='avatar_button'>{label}
        </label>
      </div>
    )
  }
}

Avatar.defaultProps = {
  label: '',
  onClick: val => val
}

Avatar.propTypes = {
  /** Label will be used for displaying text on avatar */
  label: PropTypes.string,
  /** Function to be called after clicking avatar */
  onClick: val => val
}

export default Avatar
