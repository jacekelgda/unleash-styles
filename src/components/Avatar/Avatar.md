Avatar component:

    <Avatar imgSrc="http://s3.amazonaws.com/37assets/svn/765-default-avatar.png" />
