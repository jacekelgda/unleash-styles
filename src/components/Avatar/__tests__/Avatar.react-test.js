import React from 'react'
import { shallow } from 'enzyme'

import Avatar from '../Avatar'

test('Avatar renders correctly', () => {
  const wrapper = shallow(<Avatar label="test" />)
  expect(wrapper).toMatchSnapshot()
})
