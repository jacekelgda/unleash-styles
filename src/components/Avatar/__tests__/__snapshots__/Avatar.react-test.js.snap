// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <input
        className="AvatarButton"
        id="avatar_button"
        onClick={[Function]}
        type="checkbox"
    />
    <label
        className="AvatarButtonLabel"
        for="avatar_button"
    >
        test
    </label>
</div>,
  "nodes": Array [
    <div>
      <input
            className="AvatarButton"
            id="avatar_button"
            onClick={[Function]}
            type="checkbox"
      />
      <label
            className="AvatarButtonLabel"
            for="avatar_button"
      >
            test
      </label>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Avatar
        label="test"
        onClick={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Avatar {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "label": "test",
          "onClick": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <input
                    className="AvatarButton"
                    id="avatar_button"
                    onClick={[Function]}
                    type="checkbox"
          />
          <label
                    className="AvatarButtonLabel"
                    for="avatar_button"
          >
                    test
          </label>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <input
                    className="AvatarButton"
                    id="avatar_button"
                    onClick={[Function]}
                    type="checkbox"
          />
          <label
                    className="AvatarButtonLabel"
                    for="avatar_button"
          >
                    test
          </label>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Avatar
    label="test"
    onClick={[Function]}
/>,
}
`;
